optin/src/swam/optin/coverage/CoverageReporter.scala

```diff
@@ -0,0 +1,82 @@
+package swam
+package optin
+package coverage
+
+/**
+  * Provides classes for dealing with Coverage information of WebAssembly information.
+  *
+  *  ==Overview==
+  *
+  * */
+import kantan.csv._
+import kantan.csv.ops._
+
+import scala.collection.mutable.ListBuffer
+
+import swam.runtime._
+import runtime.internals.interpreter._
+import swam.runtime.internals.compiler.CompiledFunction
+import swam.binary.custom.FunctionNames
+
+import cats.implicits._
+import cats.effect._
+import cats.effect.IO
+
+import java.nio.file._
+import java.util.logging._
+import java.nio.file.Path
+import java.io.File
+
+case class ModuleCoverageInfo(methodName: String, coveredInst: Long, totalInst: Long)
+
+object CoverageReporter {
+
+  def buildCoverage(listener: CoverageListener[IO]): List[ModuleCoverageInfo] = {
+
+    val covList = listener.coverageMap
+      .groupBy {
+        case (_, (name, _)) => name
+      }
+      .toList
+      .map {
+        case (name, map) => ModuleCoverageInfo(name, map.count(t => t._2._2 > 0), map.size)
+      }
+
+    covList
+  }
+
+  private def logCoverage(dir: Any, watOrWasm: Path, list: List[ModuleCoverageInfo]): Unit = {
+
+    implicit val modEncoder: HeaderEncoder[ModuleCoverageInfo] =
```

Comment:
You should use fs2 to output to a file, take a look here https://github.com/KTH/swam/blob/8f796a2cd169a1d10cb0313370b2c1635b5123b9/generator/src/swam/generator/ImportGenerator.scala#L113

Reply:
Sure, will remove the code related to Kantan.csv API and print the CSV using fs2.